MapsetVerifier.Checks/Taiko/Settings/CheckDiffSettings.cs

```diff
@@ -0,0 +1,286 @@
+﻿using MapsetVerifier.Parser.Objects;
+using MapsetVerifier.Framework.Objects;
+using MapsetVerifier.Framework.Objects.Attributes;
+using MapsetVerifier.Framework.Objects.Metadata;
+
+namespace MapsetVerifier.Checks.Taiko.Settings;
+
+[Check]
+public class CheckDiffSettings : BeatmapCheck
+{
+    // private const bool _DEBUG_SEE_ALL_HP = false;
+    // private const bool _DEBUG_SEE_ALL_OD = false;
+    private readonly Beatmap.Difficulty[] difficulties =
+    [
+        Beatmap.Difficulty.Easy,
+        Beatmap.Difficulty.Normal,
+        Beatmap.Difficulty.Hard,
+        Beatmap.Difficulty.Insane,
+    ];
+    
+    private static double NormalizeHpWithDrain(double hp, double drain)
```

Comment:
It seems odd that this is a function in the first place. It would be much easier to understand this if we had a nested dictionary where we could get the HP value directly from instead of doing a calculation.

So, having a sort of table layout where you create the dictionary:
```
{
   Easy: [4, 3, 2, 1],
   Normal: [5, 4, 3, 2] // or whatever values Taiko uses
}
```

So you can call it something like `recommendedHp[difficulty][drainIndex]`
or something along those lines (so short songs get index 0, middle-length songs get index 1, etc.)

You could check the CheckSpinnerGap class, which has a similar kind of array approach (while being a bit simpler)